Verify MD5 checksums returned by SQS

tawan opened the pull request:
Verify MD5 checksums returned by SQS for received messages.
 * Verify checksum of message body.
 * Verify checksum of message attributes if present.

 Refer to #1085

trevorrowe commented:
I took a moment to review this. The code looks good. I had one piece of feedback above we need to address and then I'm happy to merge this in. Good work!

tawan commented on aws-sdk-core/lib/aws-sdk-core/plugins/sqs_md5s.rb:
```diff
@@ -0,0 +1,148 @@
+            attributes = entry[:message_attributes]
+            message_response = response.successful.select { |r| r.id == id }[0]
+            unless message_response.nil?
+              validate_single_message(body, attributes, message_response)
```
@trevorrowe This is the line you mentioned in the previous commit: I simple added a `nil` check. Only messages which are which are placed in the `successful` array are verified.

tawan commented:
@trevorrowe Thanks for your feedback. I incorporated your suggestion. I amended my changes and rebased, therefore your inline comment seems to be gone. I commented the line in question again for some context.

trevorrowe commented:
Thanks for the contribution! This should go out with our next release. 

I'm going to do a small amount of refactoring on the tests. We've been moving away from unit testing the handlers in isolation and instead functionally test the client class they are applied to. This has the benefit of ensuring the plugin is applied correctly. Your tests are organized well enough that moving this should be pretty straightforward.
